src/linear_algebra/matrix/symmetric.lean

```diff
@@ -0,0 +1,115 @@
+/-
```

Comment:
I think this might be a better fit for `data/matrix/symmetric`

Reply:
I used the location other people suggested on Zulip. Do I need to change?

Reply:
Let's see what @jcommelin thinks.

Reply:
By the way, the two coming new files are `circulant.lean` and `diagonal.lean`. I think it makes sense to put them in the same place.

Reply:
`symmetric.lean` is blocking some parts of that two helpers.

Reply:
I think it's fine to put this in `linear_algebra`. It's not adding new data structures, which is what `data/` was originally meant to be for.

Reply:
My thought was that if we don't need to import `linear_algebra.basic` then it probably doesn't belong in linear_algebra.

Either way, moving files later if we change our minds is cheap - so I'm happy to have it stay here for now.